docs/reference/module-api/module-overview.md

```diff
@@ -689,23 +714,19 @@ module.exports = {
 };
 ```
 
-### `routes(self)`
+If including a route option, use an object. See the [route options](#route-options) section for more.
 
-Add standard Express routes. The `routes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those properties should be set to an object of functions. Route functions may be asynchronous (async).
+### `routes(self)`
 
-Each `routes` function takes the Express arguments `req` (the [request object](https://expressjs.com/en/api.html#req)) and `res` (the [response object](https://expressjs.com/en/api.html#res)). The functions must generate a response via `res` to avoid leaking resources, typically using the `res.redirect` or `res.send` methods.
+Add standard Express routes. The `routes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those HTTP verb properties should be set to an object of routes.
 
-See [Naming routes](#naming-routes) for more on function names and their route URLs.
-
-::: tip
-We recommend using `apiRoutes` or `restApiRoutes` whenever possible before using `routes` as they handle the potential pitfalls of Express routes. There are situations where writing Express routes may be necessary, such as when you need to use `res.redirect` or pipe a stream.
-:::
+Each route can be defined as a function or an object. If not using any route options, use a function that accepts a request as its argument:
 
 ```javascript
 // modules/product/index.js
 module.exports = {
   // ...
-  renderRoutes(self) {
+  routes(self) {
     return {
       get: {
         // GET /api/v1/product/redirect/:_id
```

Comment:
There's no `:_id` here. That sounds like a copy and paste from restApiRoutes which is the only place that parameter would ever be implied. As written this route would ned to get any arguments from `req.query`. However you could write:

```
'redirect/:_id': async (req, res) => { ... }
```

If you wanted a URL-based parameter to populate in `req.params._id`.